packages/material-ui/src/Breadcrumbs/BreadcrumbCollapsed.js

```diff
@@ -1,52 +1,114 @@
+import useThemeProps from '../styles/useThemeProps';
 import { emphasize } from '../styles/colorManipulator';
 import MoreHorizIcon from '../internal/svg-icons/MoreHoriz';
 import ButtonBase from '../ButtonBase';
+import breadcrumbCollapsedClasses, {
+  getBreadcrumbCollapsedUtilityClass,
+} from './breadcrumbCollapsedClasses';
 
-const styles = (theme) => ({
-  button: {
-    display: 'flex',
-    marginLeft: theme.spacing(0.5),
-    marginRight: theme.spacing(0.5),
-    backgroundColor: theme.palette.grey[100],
-    color: theme.palette.grey[700],
-    borderRadius: 2,
-    '&:hover, &:focus': {
-      backgroundColor: theme.palette.grey[200],
-    },
-    '&:active': {
-      boxShadow: theme.shadows[0],
-      backgroundColor: emphasize(theme.palette.grey[200], 0.12),
-    },
+const overridesResolver = (props, styles) => {
+  return deepmerge(styles.root || {}, {
+    [`& .${breadcrumbCollapsedClasses.button}`]: styles.button,
+    [`& .${breadcrumbCollapsedClasses.icon}`]: styles.icon,
+  });
+};
+
+const useUtilityClasses = (styleProps) => {
+  const { classes } = styleProps;
+
+  const slots = {
+    root: ['root'],
+    button: ['button'],
+    icon: ['icon'],
+  };
+
+  return composeClasses(slots, getBreadcrumbCollapsedUtilityClass, classes);
+};
+
+const BreadcrumbCollapsedRoot = experimentalStyled(
+  'li',
+  {},
+  {
+    name: 'PrivateBreadcrumbCollapsed',
+    slot: 'Root',
+    overridesResolver,
+  },
+)({});
+
+const BreadcrumbCollapsedButton = experimentalStyled(
+  ButtonBase,
+  {},
+  {
+    name: 'PrivateBreadcrumbCollapsed',
+    slot: 'Button',
   },
-  icon: {
-    width: 24,
-    height: 16,
+)(({ theme }) => ({
+  display: 'flex',
+  marginLeft: theme.spacing(0.5),
+  marginRight: theme.spacing(0.5),
+  backgroundColor: theme.palette.grey[100],
+  color: theme.palette.grey[700],
+  borderRadius: 2,
+  '&:hover, &:focus': {
+    backgroundColor: theme.palette.grey[200],
   },
-});
+  '&:active': {
+    boxShadow: theme.shadows[0],
+    backgroundColor: emphasize(theme.palette.grey[200], 0.12),
+  },
+}));
+
+const BreadcrumbCollapsedIcon = experimentalStyled(
+  MoreHorizIcon,
+  {},
+  {
+    name: 'PrivateBreadcrumbCollapsed',
+    slot: 'Icon',
+  },
+)(() => ({
+  width: 24,
+  height: 16,
+}));
 
 /**
  * @ignore - internal component.
  */
-function BreadcrumbCollapsed(props) {
-  const { classes, ...other } = props;
+const BreadcrumbCollapsed = React.forwardRef(function BreadcrumbCollapsed(inProps, ref) {
+  const props = useThemeProps({ props: inProps, name: 'PrivateBreadcrumbCollapsed' });
+  const { ...other } = props;
+
+  const styleProps = { ...props };
+
+  const classes = useUtilityClasses(styleProps);
 
   return (
-    <li>
-      <ButtonBase className={classes.button} focusRipple {...other}>
-        <MoreHorizIcon className={classes.icon} />
-      </ButtonBase>
-    </li>
+    <BreadcrumbCollapsedRoot className={classes.root} styleProps={styleProps}>
```

Comment:
@oliviertassinari should we leave this as `<li />`? It feels awkward as it is the root, but I don't see a point of changing it for a component.

Reply:
Also, seems like a big that className is not appended here, or if it should be on the button, then we should contact it with the `classes.button`

Reply:
The component is private
```suggestion
    <li>
```